corehq/motech/value_source.py

```diff
@@ -27,32 +30,34 @@
 from .serializers import serializers
 from .utils import simplify_list
 
+T = TypeVar('T')
 
-@attr.s
+
+@dataclass
 class CaseTriggerInfo:
-    domain = attr.ib()
-    case_id = attr.ib()
-    type = attr.ib(default=None)
-    name = attr.ib(default=None)
-    owner_id = attr.ib(default=None)
-    modified_by = attr.ib(default=None)
-    updates = attr.ib(factory=dict)
-    created = attr.ib(default=None)
-    closed = attr.ib(default=None)
-    extra_fields = attr.ib(factory=dict)
-    form_question_values = attr.ib(factory=dict)
+    domain: str
+    case_id: str
+    type: Optional[str] = None
+    name: Optional[str] = None
+    owner_id: Optional[str] = None
+    modified_by: Optional[str] = None
+    updates: dict[str, Any] = field(default_factory=dict)
+    created: Optional[bool] = None
+    closed: Optional[bool] = None
+    extra_fields: dict[str, Any] = field(default_factory=dict)
+    form_question_values: dict[str, Any] = field(default_factory=dict)
```

Comment:
No, the switch wasn't required, but it helps mypy to identify the types of fields. In the typing-extensions library and in Python 3.11, [PEP 681](https://peps.python.org/pep-0681/) is adding more support for authors of libraries like attrs, pydantic, SQLAlchemy and Django ORMs, etc. so that type checkers can check field types.